### PR TITLE
pull from feature/model_packing

### PR DESCRIPTION
**Pull request by rllin-fathom**
https://app.asana.com/0/1137246510213018/1141210948314505/f

paired with https://github.com/medicode/diseaseTools/pull/4905

1. for tpu (and packed gpu), we ensure that tensors are padded to a fixed length
2. nothing changes for gpu and chunked gpu runs

### CODE REVIEW

File: tensor2tensor/data_generators/problem.py [hunk 1]
```diff
@@ -951,12 +952,32 @@ def define_shapes(example):
...
-        # examples across all datashards
-        batch_size = params["batch_size"]
+        tf.logging.info(f'Padding features for fixed inputs: {padded_shapes}')
+
+        # on TPU, params["batch_size"] is assigned in
+        # https://github.com/medicode/tensor2tensor/blob/1525870c3a8ebc37240824a87532328e31d66887/tensor2tensor/utils/trainer_lib.py#L270
+        # to specify the number of examples for all datashards
+        # on GPU, num_shards specifies the number datashards and, thus in the
+        # packed case, also the number of examples for all datashards
+        batch_size = params.get('batch_size', num_shards)
```
Comment (etragas-fathom) [inline tensor2tensor/data_generators/problem.py]: This should go on line 977 right?

File: tensor2tensor/data_generators/problem.py [hunk 2]
```diff
@@ -1340,6 +1412,9 @@ def standardize_shapes(features, batch_size=None):
       # Assert shapes are fully known
       t.get_shape().assert_is_fully_defined()
 
+  for n, t in features.items():
+      tf.logging.info(f'Shape for {n}: {t}, {t.get_shape().as_list()}')
```
Comment (etragas-fathom) [inline tensor2tensor/data_generators/problem.py]: 'Standardized Shape'

File: tensor2tensor/data_generators/problem_test.py [hunk 3]
```diff
@@ -145,7 +146,8 @@ def testChunkPadding(self):
       example = {'inputs': ex_1_inputs,
                'targets': ex1_targets}
 
-      padded_example = sess.run(pad_inputs_to_chunk_len(example, chunk_size))
+      padded_example = sess.run(
+        pad_to_next_chunk_length(chunk_size, axis=0)(example))
```
Comment (etragas-fathom) [inline tensor2tensor/data_generators/problem_test.py]: This is missing an arg isn't it?

File: tensor2tensor/data_generators/problem.py (hunk 1 above)
Comment (rllin-fathom) [reply tensor2tensor/data_generators/problem.py]: actually should be removed.  the if else takes care of this more explicitly

File: tensor2tensor/data_generators/problem_test.py (hunk 3 above)
Comment (rllin-fathom) [reply tensor2tensor/data_generators/problem_test.py]: yes :( fixed

File: tensor2tensor/data_generators/problem.py (hunk 2 above)
Comment (rllin-fathom) [reply tensor2tensor/data_generators/problem.py]: done

**Pull request merged**